File: src/main/java/fr/cs/sio/moviesapi/model/Person.java
package fr.cs.sio.moviesapi.model;

import java.util.EnumSet;

public class Person {

    private long id;
    private String firstName;
    private String lastName;

    private EnumSet<Role> roles;

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getFirstName() {
        return firstName;
    }

    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public void setLastName(String lastName) {
        this.lastName = lastName;
    }

    public EnumSet<Role> getRoles() {
        return roles;
    }

    public void setRoles(EnumSet<Role> roles) {
        this.roles = roles;
    }
}

File: src/main/java/fr/cs/sio/moviesapi/model/Person.kt
package fr.cs.sio.moviesapi.model

import java.util.EnumSet

/**
 * A model class to represent a person.
 * Persons can be part of the cast (actors) and/or the crew (directors, etc.) of a movie.
 */
data class Person(

    val id: Long = 0L,

    val firstName: String? = null,

    val lastName: String? = null,

    // This information is *redundant* with the presence or absence of this Person in the lists of a movie.
    // But we can keep it if it adds value to our API clients.
    // Maybe we will not *store* it, but instead compute it from other relations.
    val roles: EnumSet<Role>? = null

)